frontend/src/pages/LandingPage.tsx

```diff
@@ -1,14 +1,20 @@
 // import React from "react";
+import { useRef } from "react";
 import Footer from "../components/Footer";
 // import Header from "../components/Header";
 import "../styles/LandingPage.css";
+import { useNavigate } from "react-router-dom";
+
 
 const LandingPage = () => {
+
+  const navigate = useNavigate();
+  const searchBar = useRef<HTMLInputElement>(null);
+
   const loadAccountPage = () => {
-    const rsn = (document.querySelector("#rsn") as HTMLInputElement | null)
-      ?.value;
-    window.history.pushState("", "", `/${rsn}`);
-    window.history.go();
+    const rsn = searchBar?.current?.value;
+    const path = rsn?.replace(" ", "%20");
+    navigate(`${path}`);
```

Comment:
we should probably find a package that sanitizes `rsn` for us, but we can do that later